frontend/src/components/transactions/ActionMessage.tsx

```diff
@@ -122,25 +121,36 @@ const transactionMessageRenderers: TransactionMessageRenderers = {
   AddKey: ({ transaction: { receiverId }, actionArgs }: Props<T.AddKey>) => (
     <>
       {typeof actionArgs.access_key.permission === "object" ? (
-        <>
-          {"Access key added for contract "}
-          <AccountLink
-            accountId={
-              actionArgs.access_key.permission.FunctionCall.receiver_id
-            }
-          />
-          {`: ${actionArgs.public_key.substring(0, 15)}...`}
-          <p>
-            {`with permission to call ${
-              actionArgs.access_key.permission.FunctionCall.method_names
-                .length > 0
-                ? `(${actionArgs.access_key.permission.FunctionCall.method_names.join(
-                    ", "
-                  )})`
-                : "any"
-            } methods and nonce ${actionArgs.access_key.nonce}`}
-          </p>
-        </>
+        actionArgs.access_key.permission.permission_kind ? (
```

Comment:
Is it some sort of Indexer support? Let's try to address the distinction in the `libraries/explorer-wamp/*` scope, so the rest of the code is not polluted with the changes from the switch of the data source. Please, apply the same reasoning throughout the migration.

Reply:
the current args in database: 
`{"public_key":"ed25519:F5zJLKH4V5Lyw9CvAtc6UVh7n5YhPaFKhdn2ZyzsGivL","access_key":{"nonce":0,"permission":"FullAccess"}}`

the args in indexer: 
`{"access_key": {"nonce": 0, "permission": {"permission_kind": "FULL_ACCESS"}}, "public_key": "ed25519:Ff93hQx7HBfPrAZ4uHgFBTWezB53NyrpG5SM8C2G1V2"}`

it is not straightforward to address the distinction here

Reply:
@frol